bpf/bpf_host.c

```diff
@@ -1308,12 +1308,46 @@ int cil_from_host(struct __ctx_buff *ctx)
 	return handle_netdev(ctx, true);
 }
 
+#if defined(ENABLE_IPSEC) && defined(ENABLE_IPV4) && defined(TUNNEL_MODE)
+/*
+ * If the traffic is indeed overlay traffic and it should be encrypted
+ * CTX_ACT_REDIRECT is returned, unless an error occurred, and the caller can
+ * return this code to TC.
+ *
+ * CTX_ACT_OK is returned if the traffic should continue normal processing.
+ *
+ * IS_ERR can be used to determine if this function ran into an error.
+ */
+static __always_inline int do_encrypt_overlay(struct __ctx_buff *ctx,
+					      __u16 proto)
+{
+	int ret = CTX_ACT_OK;
+	struct iphdr __maybe_unused *ipv4;
+	void __maybe_unused *data, *data_end = NULL;
+
+	/* only support vxlan for now */
```

Comment:
Could we have a good-first-issue for GENEVE support? It's easy to do and the sort of small feature gap that makes for an awkward user experience.

Reply:
Yup, Joe pointed this out as well, we can open this ticket after merge.